src/NuGet.Core/NuGet.Build.Tasks.Pack/GetProjectReferencesFromAssetsFileTask.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using NuGet.Commands;
+using NuGet.Common;
+using NuGet.ProjectModel;
+
+namespace NuGet.Build.Tasks.Pack
+{
+    public class GetProjectReferencesFromAssetsFileTask : Task
+    {
+        [Required]
+        public string RestoreOutputAbsolutePath { get; set; }
+
+        /// <summary>
+        /// Output items
+        /// </summary>
+        [Output]
+        public ITaskItem[] ProjectReferences { get; set; }
+
+        public override bool Execute()
+        {
+            // Load the assets JSON file produced by restore.
+            var assetsFilePath = Path.Combine(RestoreOutputAbsolutePath, LockFileFormat.AssetsFileName);
```

Comment:
This would work for now. Going forward we plan to make this configurable. I suggest passing in `ProjectAssetsFile` as a property, if it was passed use that. Otherwise fall back to building the default path.